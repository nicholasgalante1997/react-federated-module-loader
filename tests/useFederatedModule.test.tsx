import React from 'react';
import renderer from 'react-test-renderer';
import { useFederatedModule } from '../src/custom-hooks/useFederatedModule';


const MockComponent = () => {
    const url = 'http://localhost:9997/remoteEntry.js';
    const scope = 'mockModFedServer';
    const module = './Data';
    const { data, error, isLoading} = useFederatedModule(url, scope, module);

    if (isLoading) return <div>loading</div>;

    if (error) return <div>error</div>;

    return (
        <div>{JSON.stringify(data)}</div>
    );
};

describe('use federated module test suite', () => {
    test('it loads remote data', (done) => {
        const component = renderer.create(<MockComponent />);
        
        setTimeout(() => {
            expect(component.toJSON()).toMatchSnapshot();
            done();
        }, 3000);

    })
})