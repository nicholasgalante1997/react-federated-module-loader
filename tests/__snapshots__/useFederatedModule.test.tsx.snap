// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`use federated module test suite it loads remote data 1`] = `
<div>
  loading
</div>
`;
